auth/cache/pat.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package cache
+
+import (
+	"context"
+	"time"
+
+	"github.com/absmach/supermq/auth"
+	"github.com/absmach/supermq/pkg/errors"
+	repoerr "github.com/absmach/supermq/pkg/errors/repository"
+	"github.com/redis/go-redis/v9"
+)
+
+const Separator = "_"
+
+type patCache struct {
+	client   *redis.Client
+	duration time.Duration
+}
+
+func NewPatsCache(client *redis.Client, duration time.Duration) auth.Cache {
+	return &patCache{
+		client:   client,
+		duration: duration,
+	}
+}
+
+func (pc *patCache) Save(ctx context.Context, scopes []auth.Scope) error {
+	for _, sc := range scopes {
+		key := GenerateKey(sc.PatID, sc.OptionalDomainID, sc.EntityType, sc.Operation, sc.EntityID)
+		if err := pc.client.Set(ctx, key, true, pc.duration).Err(); err != nil {
+			return errors.Wrap(repoerr.ErrCreateEntity, err)
+		}
```

Comment:
Lets add entityID to key  and value as only `True`
`pc.client.Set(ctx, key, true, 0).Err()`

Reply:
Let's have pc.Duration, Don't set to 0, I just gave an example to for key